### ISSUE TITLE
Cannot abstract over erased calsses

### ISSUE THREAD

**Issue by nicolasstucki** [lampepfl/dotty#13428]
## Compiler version

3.0.1

## Minimized example

```Scala
import language.experimental.erasedDefinitions

erased class Capability

def f(using Capability): Int = 1
def g[C <: Capability](using C): Int = 1
```

## Output

```scala
6 |def g[C <: Capability](using C): Int = 1
  |                             ^
  |illegal reference to erased trait Capability in definition that is not itself erased
```

## Expectation
We should be able to abstract over erased classes.

**Comment by arturopala** [comment]
Does it work with `(using erased C)`?

**Comment by nicolasstucki** [comment]
Yes. But the erased classes are there to make it possible to avoid writing erased in the first place.

If you don't use erased classes you would write
```scala
import language.experimental.erasedDefinitions

type Capability

def f(using erased Capability): Int = 1
def g[C <: Capability](using erased C): Int = 1
```

**Comment by arturopala** [comment]
You are right, the doc says:
> If the type of a val definition or parameter is a (possibly aliased, refined, or instantiated) erased class, the definition is assumed **to be erased** itself.

**Comment by dwijnand** [comment]
I guess subtypes (`C`) of erased classes should also be implicitly erased?

**Comment by odersky** [comment]
> If the type of a val definition or parameter is a (possibly aliased, refined, or instantiated) erased class, the definition is assumed to be erased itself.

So that does not cover the use case, since it is neither aliased, nor refined or instantiated. Should it also work? I am not sure about that yet. Maybe better be conservative at first?
